File: java/src/test/java/codingdojo/IdTest.java
package codingdojo;

import org.junit.Test;

import java.util.function.Function;

import static org.junit.Assert.*;

public class IdTest {


    private final Function<String, Integer> length = String::length;
    private final Function<Integer, String> toString = Object::toString;

    @Test
    public void testCompositionLawPostComposition() {
        Id<String> stringId = new Id<>("");

        String value = stringId.map(length).map(toString).value;

        assertEquals("0", value);
    }

    @Test
    public void testCompositionLawPreComposition() {
        Id<String> stringId = new Id<>("");

        String value = stringId.map(length.andThen(toString)).value;

        assertEquals("0", value);
    }
}

File: csharp/TestProject/IdTest.cs
using Xunit;

namespace PimpedRose
{
    public class IdTest
    {
        [Fact]
        public void testCompositionLawPostComposition()
        {
            var stringId = new Id<string>("");

            string value = stringId
                .Select(x => x.Length)
                .Select(x => x.ToString())
                .Value;

            Assert.Equal("0", value);
        }

        [Fact]
        public void testCompositionLawPreComposition()
        {
            var stringId = new Id<string>("");

            string value = stringId
                .Select(x => x.Length.ToString())
                .Value;

            Assert.Equal("0", value);
        }
    }
}